Mathlib/Algebra/Opposites.lean

```diff
@@ -0,0 +1,382 @@
+/-
+Copyright (c) 2018 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+import Mathlib.Algebra.Group.Defs
+import Mathlib.Logic.Equiv.Defs
+import Mathlib.Logic.Nontrivial
+import Mathlib.Logic.IsEmpty
+
+/-!
+
+# Multiplicative opposite and algebraic operations on it
+
+In this file we define `MulOpposite α = αᵐᵒᵖ` to be the multiplicative opposite of `α`. It inherits
+all additive algebraic structures on `α` (in other files), and reverses the order of multipliers in
+multiplicative structures, i.e., `op (x * y) = op y * op x`, where `MulOpposite.op` is the
+canonical map from `α` to `αᵐᵒᵖ`.
+
+We also define `AddOpposite α = αᵃᵒᵖ` to be the additive opposite of `α`. It inherits all
+multiplicative algebraic structures on `α` (in other files), and reverses the order of summands in
+additive structures, i.e. `op (x + y) = op y + op x`, where `AddOpposite.op` is the canonical map
+from `α` to `αᵃᵒᵖ`.
+
+## Notation
+
+* `αᵐᵒᵖ = MulOpposite α`
+* `αᵃᵒᵖ = AddOpposite α`
+
+## Tags
+
+multiplicative opposite, additive opposite
+-/
+
+
+universe u v
+
+open Function
+
+/-- Multiplicative opposite of a type. This type inherits all additive structures on `α` and
+reverses left and right in multiplication.-/
+@[to_additive
+      "Additive opposite of a type. This type inherits all multiplicative structures on
+      `α` and reverses left and right in addition."]
+def MulOpposite (α : Type u) : Type u :=
+  α
+#align mul_opposite MulOpposite
```

Comment:
Could we either remove unnecessary #aligns, or add the additive versions as well? I think it's confusing to have one but not the other, and without explanation.

Reply:
removed the #aligns for now, let's hope this cause no problems later. then we have to add them again.